src/main/java/at/hannibal2/skyhanni/features/garden/inventory/PlotMenuHighlighting.kt

```diff
@@ -0,0 +1,88 @@
+package at.hannibal2.skyhanni.features.garden.inventory
+
+import at.hannibal2.skyhanni.config.features.garden.PlotMenuHighlightingConfig.PlotStatusTypes
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.features.garden.GardenPlotAPI
+import at.hannibal2.skyhanni.features.garden.GardenPlotAPI.currentSpray
+import at.hannibal2.skyhanni.features.garden.GardenPlotAPI.pests
+import at.hannibal2.skyhanni.features.garden.GardenPlotAPI.plots
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import net.minecraft.inventory.Slot
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class PlotMenuHighlighting {
+
+    private val config get() = GardenAPI.config.plotMenuHighlighting
+
+    private var highlightedPlots: MutableList<Pair<GardenPlotAPI.Plot, String>> = mutableListOf()
+
+    private fun isHighlighted(plot: GardenPlotAPI.Plot) = highlightedPlots.any { it.first == plot }
+    private fun isCurrentHighlight(plot: GardenPlotAPI.Plot, current: String) =
+        highlightedPlots.any { it.first == plot && it.second == current }
+
+    private fun handleCurrent(slot: Slot, plot: GardenPlotAPI.Plot, status: PlotStatusTypes) {
```

Comment:
This is duplicate code, please extract into another function
![image](https://github.com/hannibal002/SkyHanni/assets/24389977/af632470-dd82-4277-b29c-da1fc5d6e33a)